iot/iot-tenant-cleaner/src/main/java/io/enmasse/iot/tenantCleaner/infinispan/Application.java

```diff
@@ -0,0 +1,182 @@
+import org.infinispan.query.dsl.Expression;
+import org.infinispan.query.dsl.Query;
+import org.infinispan.query.dsl.QueryFactory;
+
+public class Application extends AbstractVerticle {
+
+    private static String TENANT_ID_TO_CLEAN;
+
+    private static String INFINISPAN_HOST;
+    private static int INFINISPAN_PORT;
+    private static String INFINISPAN_USERNAME;
+    private static String INFINISPAN_PASSWORD;
+    private static String INFINISPAN_SASL_SERVER;
+    private static String INFINISPAN_SASL_REALM;
+
+    private static RemoteCache<DeviceKey, DeviceInformation> devicesCache;
+    private static RemoteCache<DeviceConnectionKey, String> devicesConnectionCache;
+
+    public Application() {
+    }
+
+    @Override
+    public void start(Future<Void> fut) {
+
+        configure();
+        InfinispanProperties infinispanProperties= getInfinispanProperties();
+
+        DeviceManagementCacheProvider mgmtProvider = null;
+        DeviceConnectionCacheProvider deviceconProvider = null;
+        try {
+            mgmtProvider = new DeviceManagementCacheProvider(infinispanProperties);
+            mgmtProvider.start();
+            devicesCache = mgmtProvider.getDeviceManagementCache();
+
+            deviceconProvider = new DeviceConnectionCacheProvider(infinispanProperties);
+            deviceconProvider.start();
+            devicesConnectionCache = deviceconProvider.getDeviceStateCache();
+        } catch (Exception e){
+            System.err.println(String.format("[ERROR] Unable to access Infinispan caches. {}", e.getMessage()));
+            System.exit(1);
+        }
+
+        // Query and delete entries in devicesCache
+        QueryFactory queryFactory = Search.getQueryFactory(devicesCache);
+        Query query = queryFactory.from(DeviceInformation.class)
+                .select(Expression.property("deviceId"))
+                .having("tenantId")
+                .eq(TENANT_ID_TO_CLEAN)
+                .build();
+
+        List<String[]> matches = query.list();
+        matches.forEach(deviceInfo -> {
+            System.out.println(deviceInfo[0]);
+            DeviceKey key = new DeviceKey(TENANT_ID_TO_CLEAN, deviceInfo[0]);
+            devicesCache.removeAsync(key);
+
+            DeviceConnectionKey conKey = new DeviceConnectionKey(TENANT_ID_TO_CLEAN, deviceInfo[0]);
+            devicesConnectionCache.removeAsync(conKey);
+        });
+
+        // Stop the cache managers and release all resources
+        try {
+            deviceconProvider.stop();
+            mgmtProvider.stop();
+        } catch (Exception n){
+            System.err.println("[WARNING] Could not properly release cache manager resources in infinispan");
+        }
+    }
+
+    private void configure(){
+        ConfigStoreOptions yaml = new ConfigStoreOptions()
+                .setType("file")
+                .setFormat("yaml")
+                .setConfig(new JsonObject()
+                        .put("path", "/resources/config.yaml")
```

Comment:
We should make path configurable

Reply:
So make the path for the configuration file configurable ? Sounds like chicken and egg situation :)

Reply:
The path to the config file is now expected as a command line argument